tools/tpm2_createpolicy.c

```diff
@@ -0,0 +1,358 @@
+//**********************************************************************;
+// Copyright (c) 2015, Intel Corporation
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice,
+// this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice,
+// this list of conditions and the following disclaimer in the documentation
+// and/or other materials provided with the distribution.
+//
+// 3. Neither the name of Intel Corporation nor the names of its contributors
+// may be used to endorse or promote products derived from this software without
+// specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+// THE POSSIBILITY OF SUCH DAMAGE.
+//**********************************************************************;
+#include <errno.h>
+#include <limits.h>
+#include <string.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <getopt.h>
+#include <sapi/tpm20.h>
+
+#include "files.h"
+#include "log.h"
+#include "main.h"
+#include "options.h"
+#include "string-bytes.h"
+#include "tpm_session.h"
+#include "tpm_hash.h"
+
+typedef struct createpolicypcr_ctx createpolicypcr_ctx;
+struct createpolicypcr_ctx {
+    //common policy options
+    TSS2_SYS_CONTEXT *sapi_context;
+    SESSION *policy_session;
+    TPM2B_DIGEST policy_digest;
+    char policyfile[PATH_MAX];
+    bool policy_file_flag;
+    //build_pcr_policy options
+    unsigned int pcr_index;
+    TPMI_ALG_HASH hash_alg;
+    char raw_pcr_file[PATH_MAX];
+    struct {
+        bool policy_type_pcr_flag;
+        bool raw_pcr_flag;
+        bool pcr_index_flag;
+    } pcr_flags;
+};
+
+#define INIT_SIMPLE_TPM2B_SIZE( type ) (type).t.size = sizeof( type.t.buffer );
+
+#define SET_PCR_SELECT_BIT( pcr_selection, pcr ) \
+ (pcr_selection).pcrSelect[( (pcr)/8 )] |= ( 1 << ( (pcr) % 8) );
+
+TPM_RC build_pcr_policy(createpolicypcr_ctx *ctx) {
+
+    TPML_PCR_SELECTION pcrs = { 
+        .count = 1, 
+        .pcrSelections[0] = {
+            .hash = ctx->hash_alg,
+            .sizeofSelect = 3,
```

Comment:
We could read from the capability structure. Where should we place this routine? lib/pcr.c?